Parent class marked fixture no longer reached with pytest 7.2

Starting with the release of pytest 7.2, the distutils tests started failing (https://github.com/pypa/distutils/issues/186).

I read the [changelog for 7.2](https://docs.pytest.org/en/stable/changelog.html#pytest-7-2-0-2022-10-23), but I don't see anything relevant.

The issue is likely due to some of these factors:

- the [test class is a subclass](https://github.com/pypa/distutils/blob/6c39b5053c17ec41690dab9d11399e2f39389892/distutils/tests/test_build_ext.py#L87) of [another class](https://github.com/pypa/distutils/blob/6c39b5053c17ec41690dab9d11399e2f39389892/distutils/tests/support.py#L15)
- both classes are marked with `usefixtures`
- The fixture on the subclass is reliant on the prior execution of the fixture in the parent class.

See the upstream bug for:

- [x] a detailed description of the bug or problem you are having
- [x] output of `pip list` from the virtual environment you are using
- [x] pytest and operating system versions

For consideration:
- [x] minimal example if possible


I haven't put together a minimal example yet as I'm still triaging the issue. Is there a known change with pytest 7.2 that was expected to change this behavior?

Good news. A minimal repro was easy and I learned something:

```
 draft @ cat test_something.py
import pytest

@pytest.fixture
def add_attr1(request):
    request.instance.attr1 = object()


@pytest.fixture
def add_attr2(request):
    request.instance.attr2 = request.instance.attr1


@pytest.mark.usefixtures('add_attr1')
class Parent:
    pass


@pytest.mark.usefixtures('add_attr2')
class TestThings(Parent):
    def test_attrs(self):
        self.attr1
        self.attr2
 draft @ pip-run 'pytest@git+https://github.com/pytest-dev/pytest@31df38f6a79b0734be3c1386eb7df6512b1a321b^1' -- -m pytest
  WARNING: Did not find branch or tag '31df38f6a79b0734be3c1386eb7df6512b1a321b^1', assuming revision or ref.
========================================================================= test session starts ==========================================================================
platform darwin -- Python 3.11.4, pytest-7.2.0.dev331+g15ac0349, pluggy-1.2.0
rootdir: /Users/jaraco/draft
plugins: anyio-3.6.2
collected 1 item                                                                                                                                                       

test_something.py .                                                                                                                                              [100%]

========================================================================== 1 passed in 0.00s ===========================================================================
 draft @ pip-run 'pytest@git+https://github.com/pytest-dev/pytest@31df38f6a79b0734be3c1386eb7df6512b1a321b' -- -m pytest
========================================================================= test session starts ==========================================================================
platform darwin -- Python 3.11.4, pytest-7.2.0.dev338+g31df38f6, pluggy-1.2.0
rootdir: /Users/jaraco/draft
plugins: anyio-3.6.2
collected 1 item                                                                                                                                                       

test_something.py E                                                                                                                                              [100%]

================================================================================ ERRORS ================================================================================
_______________________________________________________________ ERROR at setup of TestThings.test_attrs ________________________________________________________________

request = <SubRequest 'add_attr2' for <Function test_attrs>>

    @pytest.fixture
    def add_attr2(request):
>       request.instance.attr2 = request.instance.attr1
E       AttributeError: 'TestThings' object has no attribute 'attr1'

test_something.py:10: AttributeError
======================================================================= short test summary info ========================================================================
ERROR test_something.py::TestThings::test_attrs - AttributeError: 'TestThings' object has no attribute 'attr1'
=========================================================================== 1 error in 0.02s ===========================================================================
```

The issue isn't that the fixture on the parent class isn't being called, but that the fixtures are called in a different order with the fixture on the child being called before the fixture on the parent. Independent of this specific example, I'd expect fixtures on more basic classes to be called first, as the child (derived) class is liable to depend on behavior from the parent but not vice-versa (from a modeling perspective, the parent class is largely unaware of the child classes).

--- a/changelog/10447.bugfix.rst
+++ b/changelog/10447.bugfix.rst
@@ -1,0 +1,2 @@
+markers are now considered in the reverse mro order to ensure base  class markers are considered first
+this resolves a regression.

--- a/src/_pytest/mark/structures.py
+++ b/src/_pytest/mark/structures.py
@@ -374,7 +374,9 @@ def get_unpacked_marks(
         if not consider_mro:
             mark_lists = [obj.__dict__.get("pytestmark", [])]
         else:
-            mark_lists = [x.__dict__.get("pytestmark", []) for x in obj.__mro__]
+            mark_lists = [
+                x.__dict__.get("pytestmark", []) for x in reversed(obj.__mro__)
+            ]
         mark_list = []
         for item in mark_lists:
             if isinstance(item, list):

--- a/testing/test_mark.py
+++ b/testing/test_mark.py
@@ -1130,6 +1130,41 @@ def test_mark_mro() -> None:
 
     all_marks = get_unpacked_marks(C)
 
-    assert all_marks == [xfail("c").mark, xfail("a").mark, xfail("b").mark]
+    assert all_marks == [xfail("b").mark, xfail("a").mark, xfail("c").mark]
 
     assert get_unpacked_marks(C, consider_mro=False) == [xfail("c").mark]
+
+
+# @pytest.mark.issue("https://github.com/pytest-dev/pytest/issues/10447")
+def test_mark_fixture_order_mro(pytester: Pytester):
+    """This ensures we walk marks of the mro starting with the base classes
+    the action at a distance fixtures are taken as minimal example from a real project
+
+    """
+    foo = pytester.makepyfile(
+        """
+        import pytest
+
+        @pytest.fixture
+        def add_attr1(request):
+            request.instance.attr1 = object()
+
+
+        @pytest.fixture
+        def add_attr2(request):
+            request.instance.attr2 = request.instance.attr1
+
+
+        @pytest.mark.usefixtures('add_attr1')
+        class Parent:
+            pass
+
+
+        @pytest.mark.usefixtures('add_attr2')
+        class TestThings(Parent):
+            def test_attrs(self):
+                assert self.attr1 == self.attr2
+        """
+    )
+    result = pytester.runpytest(foo)
+    result.assert_outcomes(passed=1)

